applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.h

```diff
@@ -0,0 +1,320 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Inigo Lopez, Marc Nuñez and Riccardo Rossi
+//
+
+#if !defined(KRATOS_TRANSONIC_PERTURBATION_POTENTIAL_FLOW_ELEMENT_H)
+#define KRATOS_TRANSONIC_PERTURBATION_POTENTIAL_FLOW_ELEMENT_H
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/element.h"
+#include "includes/kratos_flags.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/enrichment_utilities.h"
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template <int Dim, int NumNodes>
+class TransonicPerturbationPotentialFlowElement : public Element
+{
+public:
+    template <unsigned int TNumNodes, unsigned int TDim>
+    struct ElementalData
+    {
+        array_1d<double, TNumNodes> potentials, distances;
+        double vol;
+
+        BoundedMatrix<double, TNumNodes, TDim> DN_DX;
+        array_1d<double, TNumNodes> N;
+    };
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Element BaseType;
+    static constexpr int TNumNodes = NumNodes;
+    static constexpr int TDim = Dim;
```

Comment:
I'd rather keep `TNumNodes` and `TDim` (in that way it would be clear all throughout the code that these are template parameters).